tests/ui/typeck/method-chain-gats.stderr

```diff
@@ -0,0 +1,30 @@
+error[E0277]: the trait bound `<T as Base>::Base<C>: Functor<C>` is not satisfied
+  --> $DIR/method-chain-gats.rs:18:20
+   |
+LL |     input.fmap(f1).fmap(f2)
+   |                    ^^^^ the trait `Functor<C>` is not implemented for `<T as Base>::Base<C>`
+   |
+note: the method call chain might not have had the expected associated types
+  --> $DIR/method-chain-gats.rs:13:29
+   |
+LL | fn fmap2<T, A, B, C>(input: T, f1: impl Fn(A) -> B, f2: impl Fn(B) -> C) -> T::Base<C>
+   |                             ^ `Base::Base` is `<T as Base>::Base<_>` here
+...
+LL |     input.fmap(f1).fmap(f2)
+   |           -------- `Base::Base` remains `_` here
```

Comment:
useless

Reply:
We can likely just check for `_` and avoid the label then, as a quick change.

Reply:
I can either change the `select_where_possible` a few lines below my changes to `select_all_or_error` which would make this note go away or add instead add an `is_ty_var` check.

Reply:
Can you check if `select_all_or_error` will result in any currently "good" note going away too?

Reply:
I'll make CI do the dirty work...

Reply:
> I can either change the `select_where_possible` a few lines below my changes to `select_all_or_error` which would make this note go away or instead add an `is_ty_var` check.

Went with `is_ty_var` since `select_all_or_error` turned out to be too strict.